src/arch/MemoryCard/MemoryCardDriverThreaded_Folder.cpp

```diff
@@ -0,0 +1,131 @@
+﻿#include "global.h"
+#include "MemoryCardDriverThreaded_Folder.h"
+#include "RageLog.h"
+#include "RageUtil.h"
+#include "Foreach.h"
+#include "PlayerNumber.h"
+#include "MemoryCardManager.h"
+
+#include <limits.h>
+#include <stdlib.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <bitset>
+
+static int g_currentSerial = 0;
+
+MemoryCardDriverThreaded_Folder::MemoryCardDriverThreaded_Folder()
+{
+    m_LastDevices = 0;
+}
+
+MemoryCardDriverThreaded_Folder::~MemoryCardDriverThreaded_Folder()
+{
+}
+
+bool MemoryCardDriverThreaded_Folder::FolderExists(RString path)
+{
+    if (path.empty()) {
+        return false;
+    }
+
+    const char *pathname = path.c_str();
+    struct stat info;
+
+    int statRC = stat( pathname, &info );
+    if( statRC != 0 )
+    {
+        if (errno == ENOENT)  { return false; } // something along the path does not exist
+        if (errno == ENOTDIR) { return false; } // something in path prefix is not a dir
+        return false;
+    }
+
+    if( info.st_mode & S_IFDIR ) {
+        return true;
+    }
+
+    return false;
+}
+
+bool MemoryCardDriverThreaded_Folder::TestWrite( UsbStorageDevice* pDevice )
+{
+	//TODO
```

Comment:
I won't block the merge on this but it'd be nice if this todo gets sorted out in the near future :)